connectors/runner.py

```diff
@@ -192,17 +199,15 @@ async def poll(self, one_sync=False, sync_now=True):
 
                         await self._one_sync(connector, es, sync_now)
                         if one_sync:
-                            self.stop()
                             break
                 except Exception as e:
                     logger.critical(e, exc_info=True)
                     self.raise_if_spurious(e)
 
                 if not one_sync:
                     await self._sleeps.sleep(self.idling)
-                else:
-                    self.stop()
         finally:
+            self.stop()
```

Comment:
why did you move this line? this will break `--one-sync` because it will not stop anymore in the loop

Reply:
oh because you added `self.job_cleanup_task.shutdown()`
that seems ok, as long as you `break` at the end of the loop if `one_sync` is True

Reply:
Even without `self.job_cleanup_task.shutdown()`,  `self.stop()` is needed when exiting the service, that's why I moved it to `finally`

Reply:
The problem is that `stop` will toggle the flag `running` so with `--one-sync` with your change, when we `break` we will still be in the `while` loop.  I think we need to separate the stopping logic from the cleanup logic

Reply:
Oh, I see, there are 2 loops, the outer `while` loop and an inner `for` loop. I missed the 2nd one. 

IMO, we should move below check from the inner loop to the outer loop:
https://github.com/elastic/connectors-python/blob/c0c368a5906145e61786fb50bc5e5e30adc1d9bd/connectors/runner.py#L201-L202

From my understanding, `--one-sync` is to run one sync for each connector and exit, not the first connector, which is non-deterministic and not intuitive.